version: "3.9"
services:
  rabbitmq:
    image: rabbitmq:3-management-alpine
    container_name: 'rabbitmq'
    ports:
      - "15672:15672"
    volumes:
      - ~/.docker-conf/rabbitmq/data/:/var/lib/rabbitmq/
      - ~/.docker-conf/rabbitmq/log/:/var/log/rabbitmq
#    networks:
#      - rabbitmq_go_net
  server:
    container_name: API_server
    build:
      context: .
      dockerfile: Dockerfile
    env_file: .env
    restart: on-failure
    ports:
      - "127.0.0.1:80:80"
  database:
    container_name: PG_base
    image: postgres:14.1
    environment:
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
    volumes:
      - ./db:/docker-entrypoint-initdb.d/
    ports:
      - ${PG_PORT}
      - "${PG_PORT}:${PG_PORT}"
#networks:
#  rabbitmq_go_net:
#    driver: bridge